client/src/app/pages/applications/applications-table/components/manage-columns-toolbar.tsx

```diff
@@ -0,0 +1,53 @@
+import {
+  Button,
+  OverflowMenu,
+  OverflowMenuGroup,
+  OverflowMenuItem,
+  ToolbarItem,
+} from "@patternfly/react-core";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { ColumnState } from "@app/hooks/table-controls/column/useColumnState";
+import { ManageColumnsModal } from "./manage-columns-modal";
+import { ColumnsIcon } from "@patternfly/react-icons";
+
+interface ManageColumnsToolbarProps<TColumnKey extends string> {
+  columns: ColumnState<TColumnKey>[];
+  setColumns: (newColumns: ColumnState<TColumnKey>[]) => void;
+}
+
+export const ManageColumnsToolbar = <TColumnKey extends string>({
+  columns,
+  setColumns,
+}: ManageColumnsToolbarProps<TColumnKey>) => {
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <>
+      <ToolbarItem>
+        <OverflowMenu breakpoint="md">
+          <OverflowMenuGroup groupType="button" isPersistent>
+            <OverflowMenuItem isPersistent>
+              <Button
+                variant="plain"
+                onClick={() => setIsOpen(true)}
+                icon={<ColumnsIcon />}
+              ></Button>
+            </OverflowMenuItem>
+          </OverflowMenuGroup>
+        </OverflowMenu>
+      </ToolbarItem>
+      {isOpen && (
+        <ManageColumnsModal
+          onClose={() => setIsOpen(false)}
+          description={t("Selected columns will be displayed in the table.")}
+          setColumns={setColumns}
+          columns={columns}
+          saveLabel={t("Save")}
+          cancelLabel={t("Cancel")}
+          title={t("Manage Columns")}
+        />
```

Comment:
As @rszwajko mentioned offline, the `t()` calls should be keys with the text added in the translations file to be consistent with the rest of the i18next usage